wandb/__init__.pyi

```diff
@@ -416,48 +416,93 @@ def log(
     of the W&B app, or export data to visualize and explore locally, e.g. in
     Jupyter notebooks, with [our API](https://docs.wandb.ai/guides/track/public-api-guide).
 
-    In the UI, summary values show up in the run table to compare single values across runs.
-    Summary values can also be set directly with `wandb.run.summary["key"] = value`.
-
     Logged values don't have to be scalars. Logging any wandb object is supported.
-    For example `wandb.log({"example": wandb.Image("myimage.jpg")})` will log an
+    For example `run.log({"example": wandb.Image("myimage.jpg")})` will log an
     example image which will be displayed nicely in the W&B UI.
     See the [reference documentation](https://docs.wandb.com/ref/python/data-types)
     for all of the different supported types or check out our
     [guides to logging](https://docs.wandb.ai/guides/track/log) for examples,
     from 3D molecular structures and segmentation masks to PR curves and histograms.
-    `wandb.Table`s can be used to logged structured data. See our
+    You can use `wandb.Table` to log structured data. See our
     [guide to logging tables](https://docs.wandb.ai/guides/data-vis/log-tables)
     for details.
 
     Logging nested metrics is encouraged and is supported in the W&B UI.
-    If you log with a nested dictionary like `wandb.log({"train":
-    {"acc": 0.9}, "val": {"acc": 0.8}})`, the metrics will be organized into
+    When you log with a nested dictionary like `run.log({"train":
+    {"acc": 0.9}, "val": {"acc": 0.8}})`, the metrics are organized into
     `train` and `val` sections in the W&B UI.
 
-    wandb keeps track of a global step, which by default increments with each
-    call to `wandb.log`, so logging related metrics together is encouraged.
-    If it's inconvenient to log related metrics together
-    calling `wandb.log({"train-loss": 0.5}, commit=False)` and then
-    `wandb.log({"accuracy": 0.9})` is equivalent to calling
-    `wandb.log({"train-loss": 0.5, "accuracy": 0.9})`.
+    `run.log` is not intended to be called more than a few times per second.
```

Comment:
Maybe we should make it more clear in this what the advised best practice is, since it's not to call run.log more than a few times per second. I.e what you should do in case you do need to log more frequently (I know we had this written elsewhere, so there might be a good reason you suggested removing it). 

For example, adding another sentence after: 
```suggestion
    `run.log` is not intended to be called more than a few times per second. To log more frequently than that, we would recommend you log related metrics together, or aggregate a set of steps on the client side. 
```

Reply:
> To log more frequently than that

There's no way to log more frequently without performance issues (though I think `core` does significantly better here). The advice we give comes down to whether that's what the user really wants to do---usually, it makes sense to log once every N iterations of some loop. Maybe some users want incremental logging.

Let me think about how to update this.

Reply:
Modified this, lmk what you think.

Reply:
I don’t know if we want to be so explicit about usage… seems like it could be a source for confusion issues let’s not include it?

Reply:
I think if we include a warning to not log more than several times a min, we do need to explain what you're expected to in those situations. 

If we remove that disclaimer, that should be fine. 

TLDR we should either include both (disclaimer + alternative) or none

Reply:
I like the modification below that Tima added:
```
Instead, try logging once every N iterations, or collect data over multiple
    iterations and log it in a single step.
```

Reply:
sounds good, let's make it sound like a suggestion for optimal performance?

Reply:
updated replacing "Instead" with "For optimal performance"